src/metric/event.rs

```diff
@@ -41,6 +42,8 @@ pub enum Event {
         encoding: Encoding,
         /// Encoded payload.
         bytes: Vec<u8>,
+        /// Header values used by some sinks to send metadata
+        headers: Option<HashMap<Vec<u8>, Vec<u8>>>,
```

Comment:
You've had to duplicate `HashMap<Vec<u8>, Vec<u8>>` through the codebase, going to be a real pain if you decide to change any of the internal types. How about making a type alias for this?